src/tyr/route_serializer_valhalla.cc

```diff
@@ -220,6 +220,40 @@ void locations(const valhalla::Api& api, int route_index, rapidjson::writer_wrap
   writer.end_array();
 }
 
+// Serialize turn lane information
+void turn_lanes(const TripLeg& leg,
+                const DirectionsLeg_Maneuver& maneuver,
+                rapidjson::writer_wrapper_t& writer) {
+
+  // Read edge from a trip leg
+  if (maneuver.begin_path_index() == 0 || maneuver.begin_path_index() >= leg.node_size())
+    return;
+
+  auto prev_index = maneuver.begin_path_index() - 1;
+  const auto& prev_edge = leg.node(prev_index).edge();
+
+  if (prev_edge.turn_lanes_size() > 1) {
+    writer.start_array("lanes");
+
+    for (const auto& turn_lane : prev_edge.turn_lanes()) {
+      writer.start_object();
+
+      // Directions as a bit mask
+      writer("directions", static_cast<uint64_t>(turn_lane.directions_mask()));
```

Comment:
just to be pedantic maybe we should call it a mask. also can you add this to the turn by turn documentation? that way we can spell out which bits of the mask mean which direction etc.? or maybe once you have the docs written we dont need to append `_mask` to the item in the json output because the docs will say its a mask?

```suggestion
      // Directions as a bit mask
      writer("directions_mask", static_cast<uint64_t>(turn_lane.directions_mask()));
```

Reply:
When I was thinking about it, it seemed to me that it would be better to keep the JSON response short. If you add a description of the lanes in the documentation, you can just leave 'directions' here.

Reply:
yes exactly, if we document it then `directions` is sufficient